mbed_build/_internal/config/config_layer.py

```diff
@@ -43,16 +43,36 @@ def from_config_source(cls, config_source: ConfigSource, target_labels: List[str
 
         This method translates data found in ConfigSource into modifiers specific to the given target.
         """
-        modifiers_from_config = [
-            build_modifier_from_config_entry(key=key, data=data) for key, data in config_source.config.items()
-        ]
+        namespace = config_source.namespace
+        modifiers = []
+        for key, data in config_source.config.items():
+            modifiers.append(build_modifier_from_config_entry(key=_namespace(key, namespace), data=data))
 
-        modifiers_from_target_overrides: List[Callable] = []
         allowed_target_labels = ["*"] + target_labels
         for target_label, overrides in config_source.target_overrides.items():
             if target_label in allowed_target_labels:
-                modifiers_from_target_overrides.extend(
-                    build_modifier_from_target_override_entry(key=key, data=data) for key, data in overrides.items()
-                )
+                for key, data in overrides.items():
+                    modifiers.append(
+                        build_modifier_from_target_override_entry(key=_namespace(key, namespace), data=data)
+                    )
 
-        return cls(config_source=config_source, modifiers=(modifiers_from_config + modifiers_from_target_overrides))
+        return cls(config_source=config_source, modifiers=modifiers)
+
+
+def _namespace(key: str, namespace: str) -> str:
+    """Prefix configuration key with a namespace.
+
+    Namespace is ConfigSource wide, and is resolved at source build time.
+
+    It should be one of:
+    - "target"
+    - "application"
```

Comment:
How come you have chose 'application' over 'app'?

Reply:
Derp, you're right.